File: exercicios/modulo_02/exerc_01/scrabble.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int POINTERS[] = {1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3, 1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10};

int compute_score(char word);

int main(void)
{
    // Solicitar ao usuário duas palavras
    char player1[20] = "";
    printf("Player 1: ");
    fgets(player1, 20, stdin);

    char player2[20] = "";
    printf("Player 2: ");
    fgets(player2, 20, stdin);

    // Calcular os pontos de cada palavra
    int pontosPlayer1 = compute_score(player1);
    int pontosPlayer2 = compute_score(player2);

    // Imprimir o vencedor
    if (pontosPlayer1 > pontosPlayer2)
    {
        printf("Player 1 wins!");
    }
    else if (pontosPlayer2 > pontosPlayer1)
    {
        printf("Player 2 wins!");
    }
    else
    {
        printf("Tie!");
    }

    return 0;
}

int compute_score(char word)
{
    // Computa e devolve os pontos de cada palavra
    word = tolower(word);
    char letter[5] = "";
    int sum = 0;
    int tamanho = strlen(word);

    for (int i = 0; i < tamanho; i++)
    {
        strcpy(letter, word[i]);

        if (letter == "a")
        {
            sum += POINTERS[0];
        }
        else if (letter == "b")
        {
            sum += POINTERS[1];
        }
        else if (letter == "c")
        {
            sum += POINTERS[2];
        }
        else if (letter == "d")
        {
            sum += POINTERS[3];
        }
        else if (letter == "e")
        {
            sum += POINTERS[4];
        }
        else if (letter == "f")
        {
            sum += POINTERS[5];
        }
        else if (letter == "g")
        {
            sum += POINTERS[6];
        }
        else if (letter == "h")
        {
            sum += POINTERS[7];
        }
        else if (letter == "i")
        {
            sum += POINTERS[8];
        }
        else if (letter == "j")
        {
            sum += POINTERS[9];
        }
        else if (letter == "k")
        {
            sum += POINTERS[10];
        }
        else if (letter == "l")
        {
            sum += POINTERS[11];
        }
        else if (letter == "m")
        {
            sum += POINTERS[12];
        }
        else if (letter == "n")
        {
            sum += POINTERS[13];
        }
        else if (letter == "o")
        {
            sum += POINTERS[14];
        }
        else if (letter == "p")
        {
            sum += POINTERS[15];
        }
        else if (letter == "q")
        {
            sum += POINTERS[16];
        }
        else if (letter == "r")
        {
            sum += POINTERS[17];
        }
        else if (letter == "s")
        {
            sum += POINTERS[18];
        }
        else if (letter == "t")
        {
            sum += POINTERS[19];
        }
        else if (letter == "u")
        {
            sum += POINTERS[20];
        }
        else if (letter == "v")
        {
            sum += POINTERS[21];
        }
        else if (letter == "w")
        {
            sum += POINTERS[22];
        }
        else if (letter == "x")
        {
            sum += POINTERS[23];
        }
        else if (letter == "y")
        {
            sum += POINTERS[24];
        }
        else if (letter == "z")
        {
            sum += POINTERS[25];
        }

        return sum;
    }
}

File: exercicios/modulo_02/ex01/scrabble.py
def computer_score(word:str, POINTERS:list):
    sum:int = 0

    for i in range(len(word)):
        letter:str = word[i]

        if letter == 'a':
            sum += POINTERS[0]
        elif letter == 'b':
            sum += POINTERS[1]
        elif letter == 'c':
            sum += POINTERS[2]
        elif letter == 'd':
            sum += POINTERS[3]
        elif letter == 'e':
            sum += POINTERS[4]
        elif letter == 'f':
            sum += POINTERS[5]
        elif letter == 'g':
            sum += POINTERS[6]
        elif letter == 'h':
            sum += POINTERS[7]
        elif letter == 'i':
            sum += POINTERS[8]
        elif letter == 'j':
            sum += POINTERS[9]
        elif letter == 'k':
            sum += POINTERS[10]
        elif letter == 'l':
            sum += POINTERS[11]
        elif letter == 'm':
            sum += POINTERS[12]
        elif letter == 'n':
            sum += POINTERS[13]
        elif letter == 'o':
            sum += POINTERS[14]
        elif letter == 'p':
            sum += POINTERS[15]
        elif letter == 'q':
            sum += POINTERS[16]
        elif letter == 'r':
            sum += POINTERS[17]
        elif letter == 's':
            sum += POINTERS[18]
        elif letter == 't':
            sum += POINTERS[19]
        elif letter == 'u':
            sum += POINTERS[20]
        elif letter == 'v':
            sum += POINTERS[21]
        elif letter == 'w':
            sum += POINTERS[22]
        elif letter == 'x':
            sum += POINTERS[23]
        elif letter == 'y':
            sum += POINTERS[24]
        elif letter == 'z':
            sum += POINTERS[25]

        return sum


def main():
    POINTERS:list = [1, 3, 3, 2, 1, 4, 2, 4, 1, 8, 5, 1, 3, 1, 1, 3, 10, 1, 1, 1, 1, 4, 4, 8, 4, 10]
    
    # Solicitar ao usuário duas palavras
    player1:str = str(input("Player 1: ")).strip().lower()
    player2:str = str(input("Player 2: ")).strip().lower()

    # Calcular os pontos de cada palavra
    pontosPlayer1:int = computer_score(player1, POINTERS)
    pontosPlayer2:int = computer_score(player2, POINTERS)

    # Imprimir o vencedor
    if (pontosPlayer1 > pontosPlayer2) :
        print("Player 1 wins!")
    elif (pontosPlayer2 > pontosPlayer1):
        print("Player 2 wins!")
    else:
        print("Tie!")


if __name__=='__main__':
    main()